enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,377 @@
+- Users will still able to get the latest releases and the bug fixes faster and will not be affected for any delay that might can be faced to get the solutions from downstream.
+- Provide and promote the helper to the users since they can check that it is available by using the `--list-optional` flag:
+
+```sh
+$ operator-sdk bundle validate --list-optional
+NAME           LABELS                     DESCRIPTION
+operatorhub    suite=operatorframework    OperatorHub.io metadata validation
+               name=operatorhub          
+```
+
+**Cons**
+
+- Provide specific OpenShift solutions in upstream might hurt the community
+- OLM team would not like to add a code which is valid only for OpenShift unless we provide others vendor's validators as well in the public API. They are trying decouple the rules which are specific for OCP from OLM.
+- In the future it might result an effort required to move any OpenShift rules for downstream repos and implementations which are only valid for it. Currently, it would be the only case scenario so far, however, the precedence can bring others ones.
+
+### (Option B) Include the OpenShift Validator in downstream component for [operator-framework/api][oper-api] and use it in SDK upstream repository
+
+**Pros**
+
+- Mainly all pros of option A
+- Solves the OLM team concern about add OpenShift checks in [operator-framework/api][oper-api]
+  
+**Cons**
+
+- Effort required to keep a downstream repository for a downstream component and its releases [operator-framework/api][oper-api]
+- SDK would need to probably use only the downstream import to avoid misleading
+- It could cause delays for the SDK be able to apply the required changes of [operator-framework/api][oper-api] which can result ,for example, in an impact for Pipeline and users to get bug fixes or improvements.
+
+### (Option C) Design and implement Pluggable Validator mechanism for SDK
+
+Instead of we add the OpenshiftValidator to [operator-framework/api][oper-api] we would only provide it via the custom "downstream plugin" and make it available only for SDK binary.
+
+**Pros**
+
+- Address an RFE which allow others create their own validators and use them with SDK.
+- Does not affect the users and Pipeline since they will still able to use the upstream binary and the downstream validator module
+- Address the OLM team concerns about add the validator in the upstream api. Also, address similar concerns, if they exist, for SDK CLI in upstream.
+
+**Cons**
+
+- The Validator cannot be used outside of SDK. So, for example, it cannot be used by [audit proposal][audit-ep] if we decide to not make this solution leverage in SDK CLI (binary).
+- Brings higher effort to implement since requires the design and implementation of a new feature for SDK which we have none other use case/request to use that so far which justify it.
```

Comment:
While there are not specific examples aside from the existing on in this EP, having a flexible mechanism would be a good item since it could be used not only by SDK but whatever tool can import these validators. And all checks can be transformed into these dynamic checks and removed from o-f/api entirely

Reply:
It is not the idea that you are suggesting bellow. See; https://github.com/openshift/enhancements/pull/661#discussion_r598284681